src/main/java/com/verapdf/restapi/executor/Job.java

```diff
@@ -0,0 +1,171 @@
+    private final String UNABLE_TO_TRANSFER_FILE = "Unable to transfer file";
+
+    private final UUID jobId;
+
+    private Map<UUID, File> files;
+
+    private File jobDirectory;
+    private File pdfDirectory;
+
+    public Job(String rootDirectory, String pdfDirectory) {
+        files = new HashMap<>();
+        this.jobId = prepareJob(rootDirectory, pdfDirectory);
+    }
+
+    private UUID prepareJob(String rootDirectory, String pdfDirectory) {
+        UUID uuid;
+
+        do {
+            uuid = UUID.randomUUID();
+            this.jobDirectory = new File(rootDirectory, uuid.toString());
+        } while (jobDirectory.exists());
+
+        this.pdfDirectory = new File(this.jobDirectory, pdfDirectory);
+        try {
+            FileUtils.forceMkdir(this.jobDirectory);
+            FileUtils.forceMkdir(this.pdfDirectory);
+        } catch (IOException e) {
+            throw new BadRequestException(e.getMessage());
+        }
+
+        return uuid;
+    }
+
+    public JobFileDTO addSingleFile(MultipartFile file) {
+
+        String originalFilename = file.getOriginalFilename();
+
+        if (originalFilename == null) {
+            throw new ResourceNotFoundException(FILE_NOT_FOUND);
+        }
+
+        File newFile = new File(pdfDirectory, originalFilename);
+
+        if (newFile.exists()) {
+            throw new BadRequestException(FILE_ALREADY_EXISTS);
+        }
+
+        try {
+            FileUtils.copyInputStreamToFile(file.getInputStream(), newFile);
+        } catch (IOException e) {
+            LOGGER.error("Unable to transfer file.", e);
+            throw new BadRequestException(UNABLE_TO_TRANSFER_FILE);
+        }
+        UUID newFileUUID = getUniqueFileUUID();
+        files.put(newFileUUID, newFile);
+        String path = jobDirectory.toURI().relativize(newFile.toURI()).getPath();
+        return new JobFileDTO(this.jobId, newFileUUID, newFile, path, JobFileDTO.FileType.REMOTE);
+    }
+
+    public JobFileDTO addSinglePath(String path) {
+        File file = new File(path);
+        UUID newUUID;
+
+        if (!file.exists()) {
+            throw new ResourceNotFoundException(FILE_NOT_FOUND);
+        }
+
+        for (File item : files.values()) {
+            if (file.equals(item)) {
+                throw new BadRequestException(FILE_ALREADY_EXISTS);
+            }
+        }
+        newUUID = getUniqueFileUUID();
+        files.put(newUUID, file);
+
+        return new JobFileDTO(this.jobId, newUUID, file, file.getAbsolutePath(), JobFileDTO.FileType.LOCAL);
+    }
+
+    public JobFileDTO deleteFile(UUID fileId) {
+        if (!files.containsKey(fileId)) {
+            throw new ResourceNotFoundException(FILE_NOT_FOUND);
+        }
+        File file = files.get(fileId);
+        String filePath = file.getAbsolutePath();
+        JobFileDTO.FileType fileType;
+
+        if (file.exists() && pdfDirectory.equals(file.getParentFile())) {
+            fileType = JobFileDTO.FileType.REMOTE;
```

Comment:
we have a method to obtain File type (L148). Why not to use it here and 
if (filetype == REMOTE) {
    file.delete();
}